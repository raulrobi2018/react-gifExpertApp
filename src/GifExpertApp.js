import React, { useState } from "react";
import PropTypes from "prop-types";
import { AddCategory } from "./components/AddCategory";

// Utilizando el comando rafc crea el componente con su contenido inicializado
const GifExpertApp = () => {
  const [categories, setCategories] = useState([
    "One Punch",
    "Samurai X",
    "Dragon Ball",
  ]);

  // const handleAdd = () => {
  //     // Si hacemos esto está mal. Aquí lo que hace es pasar por arriba el state de categories
  //     // y lo convierte en un string
  //     // setCategories("Raul");

  //     // Extrae las categorías con el operador spread dentro del array y le agrega uno nuevo
  //     //Si se quiere agregar en primer lugar: ['Raul', ...categories]
  //     setCategories([...categories, "Raul"]);

  //     // Otra forma es retornar un callback
  //     // setCategories((cats) => [...cats, "Raul"]);
  // };

  return (
    <>
      <h2>GifExpertApp</h2>
      {/* Envía la función setCategories como props */}
      <AddCategory setCategories={setCategories} />

      <hr />
      <ol>
        {
          // Siempre se necesita tener el key de cada elemento
          // Por lo general el atributo "key" lleva un id de base de datos o un index
          // El key es usado por React para tener una referencia del elemento cuando por ejemplo
          // algo cambia en ese elemento y no tenga que renderizar todo
          // En este caso, le ponemos category porque en el array no se repiten
          categories.map((category) => {
            return <li key={category}>{category}</li>;
          })
        }
      </ol>
    </>
  );
};

export default GifExpertApp;
