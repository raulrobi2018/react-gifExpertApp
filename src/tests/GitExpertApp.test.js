import GifExpertApp from "../GifExpertApp";
import React from "react";

//Para que me de la ayuda de funciones hago esta importaciÃ³n
import "@testing-library/jest-dom";
import {shallow} from "enzyme";

describe("GifExpertApp testing", () => {
    test("should display GigExpertApp correctly", () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test("should display the list of categories (GifGrid) correctly and the length must be equal", () => {
        const categories = ["One Punch", "Dragon Ball"];
        const wrapper = shallow(
            <GifExpertApp defaultCategories={categories} />
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("GifGrid").length).toBe(categories.length);
    });
});
