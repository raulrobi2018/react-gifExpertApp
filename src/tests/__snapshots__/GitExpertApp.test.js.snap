// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp testing should display GigExpertApp correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`GifExpertApp testing should display the list of categories (GifGrid) correctly and the length must be equal 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
